partner_communication/models/communication_job.py

```diff
@@ -491,6 +512,78 @@ def reset(self):
         )
         return True
 
+    def send_by_sms(self):
+        """
+        Sends communication jobs with SMS 939 service.
+        :return: list of sms_texts
+        """
+        link_pattern = re.compile(r'<a href="([^<>]*)">([^<]*)</a>')
+        sms_medium_id = self.env.ref("sms_sponsorship.utm_medium_sms").id
+        sms_texts = []
+        for job in self.filtered(lambda j: j.state == "pending" and j.partner_mobile):
+            sms_text = job.convert_html_for_sms(link_pattern, sms_medium_id)
+            sms_texts.append(sms_text)
+            job.partner_id.with_context(
+                sms_provider=job.sms_provider_id
```

Comment:
This context won't work here. Either we can chose a provider in another way in v14, or we simply delegate the provider choice outside (meaning it cannot be set on each communication job, but in the general configuration).